app/controllers/comment_controller.rb

```diff
@@ -30,12 +34,36 @@ def create
           end
         end
       end
-    else
+    rescue CommentError
       flash[:error] = 'The comment could not be saved.'
       render text: 'failure'
     end
   end
 
+  def create_by_token
+    @node = Node.find params[:id]
+    @user = User.find_by_username params[:username]
+    @body = params[:body]
+    @token = request.env['rack.session']['token']
```

Comment:
I will look around for the best practices regarding tokens and come back to you. AFAIK, there's an important security benefit of passing the auth token inside the header, but I totally forgot what it was. :sweat_smile: